### PR TITLE
Logging for the features executor

### PR DESCRIPTION
**Pull request by noescape00**

### CODE REVIEW

File: src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs [hunk 1]
```diff
@@ -51,73 +51,89 @@ public FullNodeFeatureExecutor(IFullNode fullNode, ILoggerFactory loggerFactory)
...
         public void Dispose()
         {
+            this.logger.LogTrace("()");
+
             try
             {
                 this.Execute(feature => feature.Dispose(), true);
             }
-            catch (Exception ex)
+            catch (AggregateException)
```
Comment (bokobza) [inline src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: I don't think you have to specify `AggregateException` here.

Comment (Aprogiena) [reply src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: Agree

Comment (dangershony) [reply src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: yes also at the other method otherwise looks good

Comment (noescape00) [reply src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: Fixed

File: src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public void Initialize()
                 this.Execute(service => service.ValidateDependencies(this.node.Services));
                 this.Execute(service => service.Initialize());
             }
-            catch (AggregateException)
+            catch (Exception)
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: just catch

Comment (noescape00) [reply src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: fixed

**Pull request merged**